### PR TITLE
Backport PHPStan configuration from `master`

### PR DESCRIPTION
**Pull request by phansys**
## Subject
Backport PHPStan configuration from `master`.

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Added
- PHPStan configuration and dependencies backported from `master` branch.
```

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -1,6 +1,12 @@
 parameters:
-    ignoreErrors:
-        - # NEXT_MAJOR: Remove this issue.
-            message: "#^Call to static method getInstance\\(\\) on an unknown class Sonata\\\\EasyExtendsBundle\\\\Mapper\\\\DoctrineCollector\\.$#"
-            count: 1
-            path: src/DependencyInjection/SonataMediaExtension.php
+	ignoreErrors:
+		-
```
Comment (jordisala1991) [inline phpstan-baseline.neon:3]: Please, use spaces instead of tabs.

File: src/Provider/CdnAwareProviderInterface.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Sonata\MediaBundle\Provider;
+
+use Sonata\MediaBundle\CDN\CDNInterface;
+use Sonata\MediaBundle\Model\MediaInterface;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+interface CdnAwareProviderInterface
```
Comment (jordisala1991) [inline src/Provider/CdnAwareProviderInterface.php:22]: this is not `pedantic`.

File: phpstan-baseline.neon (hunk 1 above)
Comment (phansys) [reply phpstan-baseline.neon:3]: I didn't edit the output from the generated baseline.
I'll perform the requested replace.

File: src/Provider/CdnAwareProviderInterface.php (hunk 2 above)
Comment (phansys) [reply src/Provider/CdnAwareProviderInterface.php:22]: I'm sorry, this file is not meant to be added in this PR.
Thank you.

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Can you do the merge 3.x into master after @phansys ? I let you merge this PR

**Comment by phansys** [comment]
>I let you merge this PR

What an honor! :trollface: 
Sure, I'll create the merge.

After all, my main goal behind these changes is to fix some inconsistencies detected by PHPStan in the projects where I'm using this package, and I'd like to avoid polluting my code with annotations that belong here.

**Pull request merged**